webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAILLMService.java

```diff
@@ -0,0 +1,226 @@
+          && (systemPrompt.contains(CONTEXT_STRING_PLACEHOLDER)
+              || userPrompt.contains(CONTEXT_STRING_PLACEHOLDER))) {
+        systemPrompt = systemPrompt.replace(CONTEXT_STRING_PLACEHOLDER, nameSplit[1]);
+        userPrompt = userPrompt.replace(CONTEXT_STRING_PLACEHOLDER, nameSplit[1]);
+      }
+
+      OpenAIClient.ChatCompletionsRequest chatCompletionsRequest =
+          buildChatCompletionsRequest(
+              prompt, systemPrompt, userPrompt, prompt.getContextMessages(), true);
+
+      OpenAIClient.ChatCompletionsResponse chatCompletionsResponse =
+          openAIClient.getChatCompletions(chatCompletionsRequest).join();
+      if (persistResults) {
+        persistCheckResult(
+            textUnit,
+            repositoryId,
+            prompt,
+            chatCompletionsResponse.choices().getFirst().message().content(),
+            aiStringCheckRepository);
+      }
+      results.add(parseResponse(chatCompletionsResponse));
+    }
+  }
+
+  private AICheckResult parseResponse(
+      OpenAIClient.ChatCompletionsResponse chatCompletionsResponse) {
+    AICheckResult result;
+    String response = chatCompletionsResponse.choices().getFirst().message().content();
+    try {
+      result = objectMapper.readValue(response, AICheckResult.class);
+    } catch (JsonProcessingException e) {
+      logger.error("Error parsing AI check result from response: {}", response, e);
+      // Nothing the user can do about this, so just skip the check
+      result = new AICheckResult();
+      result.setSuccess(true);
+      result.setSuggestedFix("Check skipped as error parsing response from OpenAI.");
+    }
+    return result;
+  }
+
+  private static String getFormattedPrompt(String prompt, String sourceString, String comment) {
+    String systemPrompt = "";
+    if (prompt != null) {
+      systemPrompt =
+          prompt
+              .replace(SOURCE_STRING_PLACEHOLDER, sourceString)
+              .replace(COMMENT_STRING_PLACEHOLDER, comment);
+    }
+    return systemPrompt;
+  }
+
+  private static OpenAIClient.ChatCompletionsRequest buildChatCompletionsRequest(
+      AIPrompt prompt,
+      String systemPrompt,
+      String userPrompt,
+      List<AIPromptContextMessage> contextMessages,
+      boolean isJsonResponseType) {
+    return chatCompletionsRequest()
+        .temperature(prompt.getPromptTemperature())
+        .model(prompt.getModelName())
+        .messages(buildPromptMessages(systemPrompt, userPrompt, contextMessages))
+        .jsonResponseType(isJsonResponseType)
+        .build();
+  }
+
+  private static List<OpenAIClient.ChatCompletionsRequest.Message> buildPromptMessages(
+      String systemPrompt, String userPrompt, List<AIPromptContextMessage> contextMessages) {
+    List<OpenAIClient.ChatCompletionsRequest.Message> messages = new ArrayList<>();
+    for (AIPromptContextMessage contextMessage : contextMessages) {
+      switch (contextMessage.getMessageType()) {
```

Comment:
Shouldn't this switch be based on: "OpenAIPromptContextMessageType" (or AIPromptContextMessageType) values?